### DISCUSSION TITLE
How to correctly customize/style components in version 5?

### DISCUSSION THREAD

**Discussion by mleister97** [ant-design/ant-design#40783] [Q&A] [answered]
In design libraries like Mantine or MUI I can simply override styles by donig something like this (source https://mui.com/material-ui/customization/how-to-customize/)

```
<Slider
  defaultValue={30}
  sx={{
    width: 300,
    color: 'success.main',
    '& .MuiSlider-thumb': {
      borderRadius: '1px',
    },
  }}
/>
```

The `sx` property allows me to 

1. use the theme's token 
2. overwrite styling of specific classes (e.g. `MuiSlider-thumb` in the example). 

What is the suggested way of antd to override the styling of the components? currently i have a long css file where i have to work with `!important` everywhere. For example, I would like to overwrite the class `.ant-list-item`. This is just an example and I am aware that you can override tokens for individual components, but it doesn't cover everything.

**Comment by Yuiai01** [comment]
@mleister97 Hello, in `Antd v5` you can use `Theme` to customize the style, you can have a look at this 
https://ant.design/docs/react/customize-theme
hope it helps you.

**Reply by ashutosh887** [reply]
I want to use custom dynamic theme for each mode (light and dark). This is not given in the Docs

**Comment by mleister97** [comment]


**Reply by arvinxx** [reply]
why use `:global` ? just use nest className, it has higher priority.
```
.infoCollapse {
  border: none;

  .ant-collapse-header{

}
```

**Reply by arvinxx** [reply]
maybe you can learn how to custom component styles with CP: https://www.yuque.com/antfe/featured/durxuu94nvgvgmzq#vFlnd

**Reply by mleister97** [reply]
_I have updated the problem description._

**Comment by mleister97** [comment, marked as answer]
The answer is relatively simple - I just got it wrong. The `:where()` selector has a specificity of 0, which means you can simply overwrite the classes. if you use scss modules and want to overwrite classes that come directly from ant design, you have to use `:global(...)`. The styling of existing ant design components can therefore simply be overwritten with css/scss.

**Reply by thuynh** [reply]
@mleister97 do you think you can provide some code for this?

**Reply by ashutosh887** [reply]
Yes @mleister97 It'd be great if you share some code snippet

**Reply by iamsamiadnan** [reply]
```
.ant_dropdown_menu {
    :global {
        .ant-dropdown-menu {
            border: 1px solid red;

            .ant-dropdown-menu-item {
                border: 1px solid green;
            }
        }
    }

}
```

**Reply by manujanardhanan** [reply]
just modifying @iamsamiadnan answer for better understanding. 

<pre>
  .your-custom-class { --> your custom class for an ant element
        :global {
          color:#fff;
            .ant-upload-drag { --> class from ant design 
                border: 1px dashed #3A8E93;
            }
        }
    }
</pre>


![image](https://github.com/ant-design/ant-design/assets/5168333/dbacbb3b-9920-4f56-b776-e8a61a527acd)
